src/Core/Upgrade/UpgradeApiResourceVisitor.php

```diff
@@ -130,8 +130,8 @@ public function enterNode(Node $node)
 
             $arguments = [];
             $operations = null === $this->resourceAnnotation->itemOperations && null === $this->resourceAnnotation->collectionOperations ? null : array_merge(
-                $this->legacyOperationsToOperations($this->getLegacyOperations()),
-                $this->legacyOperationsToOperations($this->getLegacyOperations(true), true)
+                $this->legacyOperationsToOperations($this->getLegacyOperations(true), true),
+                $this->legacyOperationsToOperations($this->getLegacyOperations())
```

Comment:
this can't be the default, indeed in API Platform 3 we recommend to put the item GET operation as first one because this is the one that identifies the Resource in the first place.

Reply:
can you kindly tell me how I can avoid that the route of the "itemOperation" take also the customized routes without changing the order? i get a 404 error if i put itemoperation first.